internal/notification/smtp_notifier.go

```diff
@@ -3,50 +3,106 @@ package notification
+		password: configuration.Password,
+		sender:   configuration.Sender,
 		host:     configuration.Host,
 		port:     configuration.Port,
+		secure:   configuration.Secure,
 		address:  fmt.Sprintf("%s:%d", configuration.Host, configuration.Port),
-		sender:   configuration.Sender,
-		username: configuration.Username,
-		password: configuration.Password,
 	}
 }
 
-func (n *SMTPNotifier) authenticatedSend(recipient string, msg string) error {
-	auth := smtp.PlainAuth("", n.username, n.password, n.host)
-	err := smtp.SendMail(fmt.Sprintf("%s:%d", n.host, n.port), auth, n.sender,
-		[]string{recipient}, []byte(msg))
-	if err != nil {
-		return err
-	}
-	return nil
-}
+// Send send a identity verification link to a user.
+func (n *SMTPNotifier) Send(recipient string, subject string, body string) error {
+	msg := "From: " + n.sender + "\n" +
+		"To: " + recipient + "\n" +
+		"Subject: " + subject + "\n" +
+		"Content-Type: text/html\n" +
+		"MIME-version: 1.0;\nContent-Type: text/html; charset=\"UTF-8\";\n\n" +
+		body
 
-func (n *SMTPNotifier) unauthenticatedSend(recipient string, msg string) error {
-	// Connect to the remote SMTP server.
 	c, err := smtp.Dial(n.address)
 
 	if err != nil {
 		return err
 	}
 
+	//Do StartTLS if available (some servers only provide the auth extnesion after, and encrpytion is preferred)
+	starttls, _ := c.Extension("STARTTLS")
+	if starttls {
+		log.Debugf("SMTP server supports STARTTLS (verify cert: %t), attempting", n.secure)
+		tlsconfig := &tls.Config {
+			InsecureSkipVerify: !n.secure,
+			ServerName: n.host,
+		}
+		err := c.StartTLS(tlsconfig)
+		if err != nil {
+			return err
+		} else {
+			log.Debug("SMTP STARTTLS completed without error")
+		}
+	} else {
+		log.Debug("SMTP server does not support STARTTLS, skipping")
+	}
+
+	//Authenticate if possible
+	if n.password != "" {
+		authExtension, m := c.Extension("AUTH")
+		if authExtension {
+			log.Debugf("Config has SMTP password and server supports AUTH with the following mechanisms: %s", m)
+			mechanisms := strings.Split(m, " ")
+			var auth smtp.Auth
+
+			if starttls && utils.IsStringInSlice("PLAIN", mechanisms) {
+				auth = smtp.PlainAuth("", n.username, n.password, n.host)
+				log.Debug("SMTP server supports AUTH PLAIN, attempting")
+			} else if utils.IsStringInSlice("LOGIN", mechanisms) {
+				auth = utils.LoginAuth(n.username, n.password)
+				log.Debug("SMTP server supports AUTH LOGIN, attempting")
+			} else if utils.IsStringInSlice("CRAM-MD5", mechanisms) {
+				auth = smtp.CRAMMD5Auth(n.username, n.password)
+				log.Debug("SMTP server supports AUTH CRAM-MD5, attempting")
+			}
+			if auth == nil {
+				return errors.New("SMTP server does not advertise a AUTH mechanism that Authelia Supports")
+			}
+			err := c.Auth(auth)
+			if err != nil {
+				return err
+			} else {
+				log.Debug("SMTP AUTH completed without error")
+			}
+		} else {
+			log.Debug("SMTP config has a password provided but server does not support AUTH, skipping")
```

Comment:
As suggested, there should be an error thrown here because user provides a password which is not consumed by the server.

Reply:
Done.